src/utils/adjust-markup.js

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2019 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+const expand = require('emmet').expandAbbreviation;
+const findAndReplace = require('hast-util-find-and-replace');
+const fromDOM = require('hast-util-from-dom');
+const { JSDOM } = require('jsdom');
+const { MarkupConfig } = require('@adobe/helix-shared');
+
+/** Pleceholder variable for the generate template. */
+const PLACEHOLDER_TEMPLATE = /\$\{\d+\}/g;
+
+/**
+ * Loads the markup config into the action, if present.
+ *
+ * @param {import("../context").Context} context some param
+ * @param {import("../context").Action} action some other param
+ */
+async function loadConfig(context, action) {
+  const config = new MarkupConfig();
+  const hasFile = await config.hasFile();
+  action.markupconfig = action.markupconfig || (hasFile && config.init()) || undefined;
+}
+
+/**
+ * Returns the HTML element for the provided HTML template.
+ *
+ * @param {String} template The HTML template to use
+ *
+ * @returns {HTMLElement} the resulting HTML element including a `${0}` placeholder
+ */
+function getHTMLElement(template) {
+  const html = expand(template);
+  const doc = new JSDOM().window.document;
+  const dom = doc.createElement('div');
+  dom.innerHTML = html;
+  return dom.firstChild;
+}
+
+/**
+ * Adjust the MDAST tree according to the markup config.
+ * This is done by registering new matchers on the VDOMTransformer before the HTML
+ * is generated in the pipeline
+ *
+ * @param {Object} context the execution context
+ * @param {Object} logger the pipeline logger
+ * @param {Object} transformer the VDOM transformer
+ * @param {Object} markupconfig the markup config
+ */
+async function adjustMDAST(context, { logger, transformer, markupconfig }) {
+  if (!markupconfig || !markupconfig.markup) {
+    return;
+  }
+
+  Object.entries(markupconfig.markup)
+    .filter((entry) => entry[1].type === 'markdown')
```

Comment:
Shouldn't this be `entry.type` instead of `entry[1].type`?

Reply:
No. `.entries(…)` returns an array of length 2. `entry[0]` is the key, `entry[1]` is the value

Reply:
```suggestion
    .filter(([_, value]) => value.type === 'markdown')
```

might be a more concise way of putting it then.